corehq/apps/app_manager/models.py

```diff
@@ -2767,10 +2767,21 @@ def validate_for_build(self):
             forms = self.forms
 
             case_tag = None
+            loaded_case_types = None
             for form in forms:
                 info = self.get_module_info()
                 form_info = {"id": form.id if hasattr(form, 'id') else None, "name": form.name}
                 non_auto_select_actions = [a for a in form.actions.load_update_cases if not a.auto_select]
+                this_forms_loaded_case_types = {action.case_type for action in non_auto_select_actions}
+                if loaded_case_types is None:
+                    loaded_case_types = this_forms_loaded_case_types
+                elif loaded_case_types != this_forms_loaded_case_types:
```

Comment:
yes, both 1 & 2 are correct but harder to test for. I'm fine leaving it as is until someone complains.